.ci/pipeline_integration_test

```diff
@@ -74,25 +84,92 @@ function hf_wait_on() {
+function request_kubeconfig() {
+    rawUrl="/apis/core.gardener.cloud/v1beta1/namespaces/$1/shoots/$2/adminkubeconfig"
+    kubectl --kubeconfig=dev/sa_kubeconfig.yaml create -f <(printf '{"spec":{"expirationSeconds":%s}}' $KUBECONFIG_EXPIRY) --raw "$rawUrl" | jq -r ".status.kubeconfig" | base64 -d  > "$3"
 }
 
-function fetch_machine_class() {
-    #bringing machineclass from secret server
-    ${cli_path} config attribute --cfg-type kubernetes --cfg-name mcm-ci-az-oot-target --key machineClass > ${SOURCE_PATH}/dev/v1machineclass.json
+# Fetch *-ci-it cluster kubeconfig
+function fetch_oot_cluster_kubeconfig() {
+    NAMESPACE=garden-mcm-ci
+    request_kubeconfig "$NAMESPACE" "$TEST_SHOOT" "dev/target_kubeconfig.yaml"
+    request_kubeconfig "$NAMESPACE" "$TEST_SHOOT" "dev/control_kubeconfig.yaml"
+}
+
+function fetch_userdata(){
+    NAMESPACE='garden'
+    kubectl --kubeconfig=dev/sa_kubeconfig.yaml get shoot $TEST_SHOOT -oyaml > dev/shoot.yaml
+    SEED_NAME=$(yq eval '.status.seedName' dev/shoot.yaml)
+    request_kubeconfig $NAMESPACE $SEED_NAME "dev/seed_kubeconfig.yaml"
+    USERDATA_SECRET_NAME=$(kubectl --kubeconfig=dev/seed_kubeconfig.yaml get secrets -n shoot--mcm-ci--$TEST_SHOOT -o custom-columns=Name:.metadata.name --no-headers | grep shoot--mcm-ci)
+    kubectl --kubeconfig=dev/seed_kubeconfig.yaml get secret $USERDATA_SECRET_NAME -n shoot--mcm-ci--$TEST_SHOOT -oyaml > dev/userData.yaml
+    userData=$(yq eval '.data.userData' dev/userData.yaml)
+}
+
+function fetch_machine_class(){
+    MCC_NAME=$(kubectl --kubeconfig=dev/seed_kubeconfig.yaml get mcc -n shoot--mcm-ci--$TEST_SHOOT -o custom-columns=Name:.metadata.name --no-headers | grep shoot--mcm-ci)
+    kubectl --kubeconfig=dev/seed_kubeconfig.yaml get mcc $MCC_NAME -n shoot--mcm-ci--$TEST_SHOOT -oyaml > ${SOURCE_PATH}/dev/v1machineclass_converted.yaml
+    yq eval 'del(.metadata.creationTimestamp) | del(.metadata.finalizers) | del(.metadata.generation) | del(.metadata.resourceVersion) | del(.metadata.uid) | del(.credentialsSecretRef)'  dev/v1machineclass_converted.yaml -i
+    yq eval '.metadata.name = "test-mc-v1" | .metadata.namespace = "default" | .secretRef.name = env(TEST_CREDS_SECRET) | .secretRef.namespace = "default"' dev/v1machineclass_converted.yaml -i
+}
 
-    #convert json to yaml, so that machineclass can be parsed during integration-test
-    yq e -P ${SOURCE_PATH}/dev/v1machineclass.json > ${SOURCE_PATH}/dev/v1machineclass_converted.yaml
+function fetch_credentials(){
+   kubectl --kubeconfig=dev/sa_kubeconfig.yaml get secret $CREDENTIALS_SECRET_NAME -n $GARDEN_CORE_NAMESPACE -o yaml > dev/credentials.yaml
+   clientID=$(yq eval '.data.clientID' dev/credentials.yaml)
+   clientSecret=$(yq eval '.data.clientSecret' dev/credentials.yaml)
+   subscriptionID=$(yq eval '.data.subscriptionID' dev/credentials.yaml)
+   tenantID=$(yq eval '.data.tenantID' dev/credentials.yaml)
+
+   echo "$clientID" "$clientSecret" "$subscriptionID" "$tenantID"
```

Comment:
```suggestion
```

Reply:
sorry, had added these for debugging during auth failures.